biz.aQute.bnd.exporters/src/aQute/bnd/exporter/feature/FeatureExporter.java

```diff
@@ -0,0 +1,259 @@
+package aQute.bnd.exporter.feature;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import org.osgi.framework.Constants;
+import org.osgi.resource.Capability;
+import org.osgi.util.feature.Feature;
+import org.osgi.util.feature.FeatureBuilder;
+import org.osgi.util.feature.FeatureBundle;
+import org.osgi.util.feature.FeatureConfiguration;
+import org.osgi.util.feature.FeatureExtension;
+import org.osgi.util.feature.Features;
+import org.osgi.util.feature.ID;
+
+import aQute.bnd.annotation.plugin.BndPlugin;
+import aQute.bnd.build.Project;
+import aQute.bnd.header.Parameters;
+import aQute.bnd.osgi.FileResource;
+import aQute.bnd.osgi.Resource;
+import aQute.bnd.osgi.resource.CapReqBuilder;
+import aQute.bnd.service.export.Exporter;
+
+@BndPlugin(name = "Feature Exporter")
+public class FeatureExporter implements Exporter {
+
+	private Project	project;
+
+	private String	type;
+
+	@Override
+	public String[] getTypes() {
+
+		return new String[] {
+			"feature", "feature.json"
+		};
+	}
+
+	@Override
+	public Map.Entry<String, Resource> export(String type, final Project project, Map<String, String> options)
+		throws Exception {
+
+		this.project = project;
+		this.type = type;
+		doFeatureFile();
+		return null;
+	}
+
+	private void doFeatureFile() throws Exception {
+
+		ID aId = new ID("groupId", "artefactId", version().orElse("0.0.0-INITIAL"));
+
+		FeatureBuilder featureBuilder = Features.getBuilderFactory()
+			.newFeatureBuilder(aId);
+
+		title().ifPresent(featureBuilder::setName);
+		decription().ifPresent(featureBuilder::setDescription);
+		license().ifPresent(featureBuilder::setLicense);
+		vendor().ifPresent(featureBuilder::setVendor);
+
+		featureConfigurations()
+			.ifPresent(cs -> featureBuilder.addConfigurations(cs.toArray(FeatureConfiguration[]::new)));
+		featureBundles().ifPresent(bs -> featureBuilder.addBundles(bs.toArray(FeatureBundle[]::new)));
+		variables().ifPresent(featureBuilder::addVariables);
+		extensions().ifPresent(es -> featureBuilder.addExtensions(es.toArray(FeatureExtension[]::new)));
+
+		// capabilities().ifPresent(cap ->
+		// featureBuilder.addCapabilities(cap.toArray(FeatureCapability[]::new)));
+		// requirement().ifPresent(rs ->
+		// featureBuilder.addRequirements(rs.toArray(FeatureRequirement[]::new)));
+
+		// launcher
+		// frameworkProperies().ifPresent(featureBuilder.getFrameworkProperties()::putAll);
+
+		// FileWriter
+		String fileName = featureName(project);
+		File jsonPath = project.getTargetDir()
+			.toPath()
+			.resolve(fileName)
+			.toFile();
+
+		if (jsonPath.exists()) {
```

Comment:
File.createNewFile always creates a new file, so this is unnecessary

Reply:
Sure! Will renove this later. This is just a draft pr. Thank you a lot for reviewing but nost here will change a lot.

But here is also a position where I create a json and not a bundle and i have no clue how to push this to maven